base/iterators.jl

```diff
@@ -914,4 +915,32 @@ function next(itr::PartitionIterator, state)
     return resize!(v, i), state
 end
 
+"""
+    only(x)
+
+Returns the one and only element of collection `x`, and throws an error if the collection
+has zero or multiple elements.
+"""
+Base.@propagate_inbounds function only(x)
```

Comment:
It seems odd to mark this this way, where `@inbounds only(x)` is `first(x)`

Reply:
The oddness here comes because I want `@inbounds` to propagate to `next`, since that can have `@boundscheck`s in it.